reference/shaders-hlsl/vert/basic.vert

```diff
@@ -0,0 +1,47 @@
+struct _UBO
+{
+    float4x4 uMVP;
+};
+
+cbuffer UBO
+{
+    _UBO _16;
+};
+uniform float4 gl_HalfPixel;
+
+static float4 gl_Position;
```

Comment:
And I suppose resource variables can always be declared in global scope (unlike Metal), so there is no need for hackery there either?